### DISCUSSION TITLE
Generate midi file - empty

### DISCUSSION THREAD

**Discussion by DevJett** [CoderLine/alphaTab#1128] [Q&A] [answered]
First thank you for the great work you have done, it's awesome! I'm trying to generate midi file and download it with javascript but I got no clue why the generated midi is empty. 

I have tested almost all version

1.3 alpha
1.2.3
1.0.1

And still the same issue 

Cloud you please take a look at my source code?
[https://codesandbox.io/s/alphatab-forked-bu3c13](https://codesandbox.io/s/alphatab-forked-bu3c13)

**Comment by Danielku15** [comment, marked as answer]
I looked at your code and generally it seems fine (after uncommenting the relevant parts in `createMidiFile`). But there is a small detail to this export: 

Please note that currently there is no "Standard Midi File" (SMF) export/output possible. alphaTab uses some "Midi 2.0" messages to handle note bends. But unfortunately the Midi Association  has not standardized the "SMF 2.0" format. You can see some details here: https://github.com/CoderLine/alphaTab/issues/943 I plan to maybe attempt readding a SMF 1.0 compliant export still in Version 1.3. 

Currently I do not have really a good workaround that would be straight forward. The "Low Level APIs" primarily serve the needs within alphaTab and might not suite all outside world needs. It might become an official public API as part of https://github.com/orgs/CoderLine/projects/12/views/1?pane=issue&itemId=23948039
